src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private int maxSize = 10;
+    private int size;
+    private T[] array;
+    private T[] arrayGrow;
+
+    public ArrayList() {
+        array = (T[]) new Object[maxSize];
+    }
 
+    public void resize() {
+        if (size == maxSize) {
+            maxSize = (int) (maxSize * 1.5);
+            arrayGrow = (T[]) new Object[maxSize];
+            System.arraycopy(array, 0, arrayGrow, 0, size);
+            array = arrayGrow;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        resize();
+        for (int i = 0; i < size; i++) {
+            if ((array[i] != null && array[i] == value)
+                    || (array[i] != null && array[i].equals(value))) {
+                array[i] = value;
+                return;
+            }
+        }
+        array[size] = value;
+        size++;
+    }
 
+    @Override
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        if (index >= size + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("No such index exception");
+        }
+        resize();
+        if (index < size) {
+            T[] arrayGrow = (T[]) new Object[maxSize];
+            System.arraycopy(array, 0, arrayGrow, 0, index);
+            arrayGrow[index] = value;
+            System.arraycopy(array, index, arrayGrow, index + 1, size - index);
+            array = arrayGrow;
+            size++;
+            return;
+        }
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (array.length < size + list.size()) {
+            maxSize = (int) (maxSize * 1.5);
+            arrayGrow = (T[]) new Object[maxSize];
+            System.arraycopy(array, 0, arrayGrow, 0, size);
+            array = arrayGrow;
+        }
+        for (int i = 0; i < list.size(); i++) {
+            array[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public T get(int index) throws ArrayListIndexOutOfBoundsException {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("No such index exception");
```

Comment:
You can write a more informative message. Include the current size and index that caused trouble. It will make debugging easier :)
Also you can make it's message a string constant